### ISSUE TITLE
Question of flashatt_kernel

### ISSUE THREAD

**Issue by Charlie0804** [SiriusNEO/Triton-Puzzles-Lite#5]
Hello author, I have some questions about the implementation of flashattention. The oi coefficient in the original formula needs to be divided by di (the pytorch code is as follows), 
![flashattention](https://github.com/user-attachments/assets/6ddc4f37-8950-4f23-ad4b-d07448f2c7c0)

`for j in range(Tc):
    Kj = K_BLOCKS[j]
    Vj = V_BLOCKS[j]
    for i in range(Tr):
        Qi = Q_BLOCKS[i]
        Oi = O_BLOCKS[i]
        li = l_BLOCKS[i]
        mi = m_BLOCKS[i]

        S_ij = torch.einsum('... i d, ... j d -> ... i j', Qi, Kj)
        m_block_ij, _ = torch.max(S_ij, dim=-1, keepdims=True)
        P_ij = torch.exp(S_ij - m_block_ij)
        l_block_ij = torch.sum(P_ij, dim=-1, keepdims=True) + EPSILON
        P_ij_Vj = torch.einsum('... i j, ... j d -> ... i d', P_ij, Vj)
        
        mi_new = torch.maximum(m_block_ij, mi)
        
        li_new = torch.exp(mi - mi_new) * li  \
               + torch.exp(m_block_ij - mi_new) * l_block_ij 

        O_BLOCKS[i] = (li / li_new) * torch.exp(mi - mi_new) * Oi \
                    +(torch.exp(m_block_ij - mi_new) / li_new) * P_ij_Vj`
but I see that your triton implementation does not have this step, perhaps my understanding is not quite in place, could you please answer it~ Thank you very much

**Comment by Charlie0804** [comment]
奥奥理解了，这个是flashattentionv2的实现
